tests/chainer_tests/functions_tests/math_tests/test_basic_math.py

```diff
@@ -216,6 +216,58 @@ def test_rpow_double_backward_gpu(self):
         self.double_backward_gpu(lambda x, y: y.__rpow__(x))
 
 
+@testing.parameterize(*testing.product({
+    'shape': [(3, 2), ()],
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+}))
+class TestBinaryOpTuple(unittest.TestCase):
+
+    def setUp(self):
+        self.x1 = numpy.random.uniform(.5, 1, self.shape).astype(self.dtype)
+        self.x2 = numpy.random.uniform(.5, 1, self.shape).astype(self.dtype)
+        self.x3 = numpy.random.uniform(.5, 1, self.shape).astype(self.dtype)
+        self.gy = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        self.ggx1 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        self.ggx2 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        self.ggx3 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+
+    def check_forward(self, func, op, x1_data, x2_data, x3_data):
+        x1 = chainer.Variable(x1_data)
+        x2 = chainer.Variable(x2_data)
+        x3 = chainer.Variable(x3_data)
+        y = func(x1, x2, x3)
+        options = {}
+        if self.dtype == numpy.float16:
+            options = {'atol': 1e-4, 'rtol': 1e-3}
+        testing.assert_allclose(
+            op(op(self.x1, self.x2), self.x3), y.data, **options)
+
+    def forward_cpu(self, func, op):
```

Comment:
Indirection with this method is needed, because only adding operation can be tested.

Reply:
I'm not quite understand `Indirection`. But I will update the test code to replace `op` with `self.x1 + self.x2 + self.x3`.